diff --git a/drivers/gpu/drm/amd/display/dc/core/dc.c b/drivers/gpu/drm/amd/display/dc/core/dc.c
index 2628f16f77c19..c744aa9d830f6 100644
--- a/drivers/gpu/drm/amd/display/dc/core/dc.c
+++ b/drivers/gpu/drm/amd/display/dc/core/dc.c
@@ -6384,35 +6384,6 @@ unsigned int dc_get_det_buffer_size_from_state(const struct dc_state *context)
 	else
 		return 0;
 }
-/**
- ***********************************************************************************************
- * dc_get_host_router_index: Get index of host router from a dpia link
- *
- * This function return a host router index of the target link. If the target link is dpia link.
- *
- * @param [in] link: target link
- * @param [out] host_router_index: host router index of the target link
- *
- * @return: true if the host router index is found and valid.
- *
- ***********************************************************************************************
- */
-bool dc_get_host_router_index(const struct dc_link *link, unsigned int *host_router_index)
-{
-	struct dc *dc = link->ctx->dc;
-
-	if (link->ep_type != DISPLAY_ENDPOINT_USB4_DPIA)
-		return false;
-
-	if (link->link_index < dc->lowest_dpia_link_index)
-		return false;
-
-	*host_router_index = (link->link_index - dc->lowest_dpia_link_index) / dc->caps.num_of_dpias_per_host_router;
-	if (*host_router_index < dc->caps.num_of_host_routers)
-		return true;
-	else
-		return false;
-}
 
 /**
  * dc_get_host_router_index: Get index of host router from a dpia link
