diff --git a/drivers/gpu/drm/xe/xe_pm.c b/drivers/gpu/drm/xe/xe_pm.c
index 2e2accd76fb2..2600c936527e 100644
--- a/drivers/gpu/drm/xe/xe_pm.c
+++ b/drivers/gpu/drm/xe/xe_pm.c
@@ -392,8 +392,6 @@ int xe_pm_runtime_suspend(struct xe_device *xe)
 	xe_display_pm_runtime_suspend(xe);
 
 	if (xe->d3cold.allowed) {
-		xe_display_pm_suspend(xe, true);
-
 		err = xe_bo_evict_all(xe);
 		if (err)
 			goto out;
diff --git a/drivers/gpu/drm/xe/display/xe_display.c b/drivers/gpu/drm/xe/display/xe_display.c
index 78a884ddd499..710b1e2170c1 100644
--- a/drivers/gpu/drm/xe/display/xe_display.c
+++ b/drivers/gpu/drm/xe/display/xe_display.c
@@ -341,8 +341,6 @@ void xe_display_pm_suspend(struct xe_device *xe, bool runtime)
 
 	xe_display_flush_cleanup_work(xe);
 
-	xe_display_flush_cleanup_work(xe);
-
 	intel_dp_mst_suspend(xe);
 
 	intel_hpd_cancel_work(xe);
