diff --git a/drivers/gpu/drm/xe/xe_pm.c b/drivers/gpu/drm/xe/xe_pm.c
index 2e2accd76fb2..2600c936527e 100644
--- a/drivers/gpu/drm/xe/xe_pm.c
+++ b/drivers/gpu/drm/xe/xe_pm.c
@@ -392,8 +392,6 @@ int xe_pm_runtime_suspend(struct xe_device *xe)
 	xe_display_pm_runtime_suspend(xe);
 
 	if (xe->d3cold.allowed) {
-		xe_display_pm_suspend(xe, true);
-
 		err = xe_bo_evict_all(xe);
 		if (err)
 			goto out;
